buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ValidateRestSpecTask.java

```diff
@@ -0,0 +1,160 @@
+        JsonSchema jsonSchema;
+        FileTree jsonSchemas = Util.getJavaMainSourceResources(
+            getProject().findProject(SCHEMA_PROJECT),
+            Util.filterByPatterns(getPatternSetFactory().create(), Collections.singletonList(JSON_SCHEMA_PATTERN), null)
+        );
+        if (jsonSchemas == null || jsonSchemas.getFiles().size() != 1) {
+            throw new IllegalStateException(
+                String.format(
+                    "Could not find the schema file from glob pattern [%s] and project [%s] for JSON spec validation",
+                    JSON_SCHEMA_PATTERN,
+                    SCHEMA_PROJECT
+                )
+            );
+        }
+        File jsonSchemaOnDisk = jsonSchemas.iterator().next();
+        getLogger().debug("JSON schema for REST spec: [{}]", jsonSchemaOnDisk.getAbsolutePath());
+        SchemaValidatorsConfig config = new SchemaValidatorsConfig();
+        JsonSchemaFactory factory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+        jsonSchema = factory.getSchema(mapper.readTree(jsonSchemaOnDisk), config);
+        // validate input files
+        FileTree specs = getInputDir();
+        Map<File, Set<ValidationMessage>> errors = new HashMap<>();
+        for (File file : specs.getFiles()) {
+            if (ignore.contains(file.getName())) {
+                getLogger().info("Ignoring file [{}] due to configuration", file.getName());
+                continue;
+            }
+            // validate all files and hold on to errors for a complete report if there are failures
+            getLogger().debug("Validating REST spec [{}]", file.getName());
+            Set<ValidationMessage> validationMessages = jsonSchema.validate(mapper.readTree(file));
+            for (ValidationMessage validationMessage : validationMessages) {
+                errors.computeIfAbsent(file, k -> new HashSet<>()).add(validationMessage);
+            }
+        }
+        if (errors.isEmpty()) {
+            Files.write(getSuccessMarker().toPath(), new byte[] {}, StandardOpenOption.CREATE);
+        } else {
+            // build output and throw exception
+            StringBuilder sb = new StringBuilder();
+            sb.append("Error validating REST specification.");
+            sb.append(System.lineSeparator());
+            sb.append(String.format("Schema: %s", jsonSchemaOnDisk));
+            sb.append(System.lineSeparator());
+            errors.forEach((file, error) -> {
+                sb.append(System.lineSeparator());
+                sb.append(String.format("File: %s", file.getName()));
+                sb.append(System.lineSeparator());
+                sb.append("----------------------");
+                sb.append(System.lineSeparator());
+                error.forEach(message -> {
+                    sb.append(message);
+                    sb.append(System.lineSeparator());
+                });
+            });
+            throw new JsonSchemaException(sb.toString());
```

Comment:
Let's avoid smashing the output of this task in an exception message. Instead let's follow the example of other core Gradle tasks which is to simply log the errors in the task using `logger` and then throw an exception message that says something like "There were errors, refer to task output.". We can (and should) include a summary in the exception (ex: "Validation failed: 3 files contained 8 violations"), but details should be in task output.

For example, here's a checkstyle error: https://gradle-enterprise.elastic.co/s/ixyvqw455k6m6/failure?openFailures=WzBd&openStackTraces=WzFd#top=0

Reply:
Done. The a validation error looks like:

```
> Task :rest-api-spec:validateRestSpec FAILED
[validate JSON][ERROR][cat.thread_pool.json][$.cat.thread_pool.params.size.options[0]: does not match the regex pattern ^[a-zA-Z_]+$]

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rest-api-spec:validateRestSpec'.
> Error validating JSON. See the report at: file:/Users/jakelandis/workspace/8x/elasticsearch/rest-api-spec/build/reports/validateJson.txt
  JSON validation failed: 1 files contained 1 violations

```

The report looks like:
```
Schema: /Users/jakelandis/workspace/8x/elasticsearch/rest-api-spec/src/main/resources/schema.json
----------Validation Errors-----------
/Users/jakelandis/workspace/8x/elasticsearch/rest-api-spec/src/main/resources/rest-api-spec/api/cat.thread_pool.json: $.cat.thread_pool.params.size.options[0]: does not match the regex pattern ^[a-zA-Z_]+$
```

Reply:
I'm not sure we need the report file _and_ the log output, what do you think? To me the log output is the priority, as that's the only convenient way to debug errors in CI. And if we have all the info we need in the console, do we need to create the report file?

Reply:
I think there is marginal benefit to having a report. It allows me to use the fqn for the files without polluting the log output with long names. Also, this validation as-is should feel familiar since it like checkstyle (not sure if that is good or bad)

I have a minor preference to keep the report.

Reply:
> I have a minor preference to keep the report.

SGTM 👍

Reply:
If we are going to keep the report, let's just go ahead and ditch the success marker then. No point in having two different outputs. In the case of scenario where we found no errors, we can just say as much in the report file. So let's get rid of that and the need to extend `PrecommitTask`.

Reply:
done. https://github.com/elastic/elasticsearch/pull/55117/commits/6090fbafe059f95885918d9d6705abe663623c31